issue_processing/jira_issue_processing.py

```diff
@@ -225,6 +259,69 @@ def merge_user_with_user_from_csv(user, persons):
     return issues
 
 
+def load_issue_via_api(issues, persons):
+    """For each issue in the list the history is added via the api
+
+        :param issues: list of issues
+        :param persons: list of persons from JIRA (incl. e-mail addresses)
+    """""
+
+    log.info("Load issue information via api...")
+    jira_project = JIRA(project_url)
+
+    for issue in issues:
+
+        min_date = issue['creationDate']
+        api_issue = jira_project.issue(issue['externalId'], expand='changelog')
+        changelog = api_issue.changelog
+        histories = list()
+
+        for change in changelog.histories:
+
+            old_state, new_state, old_resolution, new_resolution = "None", "None", "None", "None"
+
+            for item in change.items:
+
+                if item.field == 'status':
+                    old_state = str(item.fromString)
+                    new_state = str(item.toString)
+
+                elif item.field == 'resolution':
+                    old_resolution = str(item.fromString)
+                    new_resolution = str(item.toString)
+
+            # only history entries that change the state or the resolution are interesting
+            if (old_state != new_state) | (old_resolution != new_resolution):
+
+                history = dict()
+                history['state'] = new_state
+                history['resolution'] = new_resolution
+                user = dict()
+                user["username"] = change.author.name
+                user["name"] = change.author.name
+                user["email"] = ""
+                history["author"] = merge_user_with_user_from_csv(user, persons)
+                date = change.created
+                d = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S.%f+0000")
+                history['date'] = d.strftime('%Y-%m-%d %H:%M:%S')
+                histories.append(history)
+
+                # updates state and resolution for issue and comments in the past
+                if (min_date <= issue['creationDate']) & (issue['creationDate'] < history['date']):
+                    issue['state'] = old_state
+                    issue['resolution'] = old_resolution
+
+                for comment in issue['comments']:
+
+                    if (min_date <= comment['changeDate']) & (comment['changeDate'] < history['date']):
```

Comment:
Same holds here: Not sure why we need to have this if-block here and what you want to achieve with it.

Reply:
Analogous to the if-block above. This Block elaborates what state and resolution the issue had when the comment was written. In my opinion this is an interesting information, because it makes a difference if you comment on an resolved bug or on an unresolved bug. If we don't want this information this if-block is unnecessary

Reply:
Thanks for giving me a rationale. :+1: The information is strongly needed, but needs to be incorporated solely in the event part of the data (see also the comments on Lines 310 and 498).